src/components/AnalysisPage/3dModelViewer/3dModelViewer.jsx

```diff
@@ -16,18 +16,34 @@ const ModelViewer = ({
 }) => {
   const canvasRef = useRef(null);
   const sceneRef = useRef(null);
+  const [previousFileFor3dModel, setPreviousFileFor3dModel] = useState(null);
+  const [
+    previousOriginalFileNameFor3dModel,
+    setPreviousOriginalFileNameFor3dModel,
+  ] = useState(null);
 
   useEffect(() => {
     sceneRef.current = setupThreeScene(canvasRef.current);
   }, []);
 
   useEffect(() => {
+    const isNewFileFor3dModel =
+      fileFor3dModel && fileFor3dModel !== previousFileFor3dModel;
+    const isNewOriginalFileNameFor3dModel =
+      originalFileFor3dModel &&
+      originalFileFor3dModel !== previousOriginalFileNameFor3dModel;
+
+    const viewingNewPart =
+      isNewFileFor3dModel && isNewOriginalFileNameFor3dModel;
```

Comment:
THis is starting to get a little confusing with all the current vs original vs previous stuff for everything. Im not quite sure how to make it better though. Maybe just refactoring to make some small well named functions or something.

Or maybe theres a way to split it apart so that instead of all the loading logic being here we could have like a function to load in a new original file and then a separate function to load the ply visualization file into the viewer.

Not sure just thinking out loud, we can keep an eye on this as we continue developing the viewer

Reply:
In one of my latest commits, I was able to completely remove all of this! 

no more confusion